cli-client/helpers/tts_client.py

```diff
@@ -24,19 +36,109 @@ def _compose_synthesis_request(self, text: str, voice: str, audio_format: str, s
-        return response.audio_samples
+        logging.info("Received response with %s bytes of audio data", len(response.audio_samples))
+        self._save_audio_result(response.audio_samples)
+
+    def _close_stream_by_inactivity(self):
+        logging.info("Stream inactivity detected, closing stream...")
+        self._peer_responded.set()
+
+    def _start_inactivity_timer(self, inactivity_timeout: float):
+        self._inactivity_timer = Timer(inactivity_timeout, self._close_stream_by_inactivity)
+        self._inactivity_timer.start()
+    
+    def _response_watcher(self, response_iterator):
+        try:
+            audio = bytearray()
+            logging.info("Running response watcher")
+            for response in response_iterator:
+                logging.debug("New incoming response of type: %s", type(response))
+                
+                if response.streaming_audio.audio_samples:
+                    logging.info("StreamingAudio response received: %s bytes of audio data", len(response.streaming_audio.audio_samples))
+                    audio.extend(response.streaming_audio.audio_samples)
+                
+                if response.end_of_utterance.data:
+                    logging.info("EndOfUtterance response received. Signaling end of stream with data: %s", response.end_of_utterance.data)
+
+                if self._inactivity_timer:
+                    self._inactivity_timer.cancel()
+                self._start_inactivity_timer(self._inactivity_timer_timeout)
+                
+            self._save_audio_result(audio)
+
+        except Exception as e:
+            logging.error("Error running response watcher: %s", str(e))
+            self._peer_responded.set()
+            raise
+    
+    def send_text(self) -> None:
+        metadata = None if self._secure_channel else [('authorization', "Bearer " + self._token)]
+        self.__generate_messages(
+                text_file=self._text_file,
+                voice=self._voice,
+                sample_rate=self._audio_sample_rate),
+        response_iterator = self._stub.StreamingSynthesizeSpeech(self.__message_iterator(), metadata=metadata)
+        self._consumer_future = self._executor.submit(self._response_watcher, response_iterator)
+
+    def wait_for_response(self) -> bool:
+        logging.info("Waiting for server to respond...")
+        self._peer_responded.wait(timeout=None)
+        if self._consumer_future.done():
+            self._consumer_future.result()
+
+        return True
+
+    def __message_iterator(self):
+        for message_type, message in self._messages:
+            logging.info("Sending streaming message " + message_type)
+            yield message
+        logging.info("All audio messages sent")
+
+    def __generate_messages(
+        self, 
+        text_file: str, 
+        voice: str,
+        sample_rate: int, 
+    ):
+        supported_sample_rates = {8000: 0, 16000: 1}
```

Comment:
I would use the enum variants' names as it's easier to understand:

```suggestion
        supported_sample_rates = {8000: VoiceSamplingRate.VOICE_SAMPLING_RATE_8KHZ, 16000: VoiceSamplingRate.VOICE_SAMPLING_RATE_16KH}
```